### ISSUE TITLE
Arduino Port

### ISSUE THREAD

**Issue by el-samiyel** [PhillyNJ/SAMD21#1]
Hi, great work. Are there any plans to have the crypto library ported to arduino. 

Is this easy enough to do?

Cheers

**Comment by PhillyNJ** [comment]
Thank you. Yes, I just finished the Arduino port but please be aware that the Arduino Uno doesn't have enough program/data size to run the examples. I used a Arduino Mega 2560 for the demo, Also, the example is for the Single Wire Version. 

Unzip the to the "libraries" folder, restart the Arduino IDE and you should see 2 examples. Read the comments in the example because you need to add a platform.local.txt and define your single wire pin

`PLEASE READ

    Requires platform.local.txt with symbols defined.
    File should be saved in {arduino home}\hardware\arduino\avr

     Create or update "{arduino}\hardware\arduino\avr\platform.local.txt"
  	compiler.c.extra_flags=-DATCA_HAL_SWI
  	compiler.c.elf.extra_flags=-DATCA_HAL_SWI
  	compiler.S.extra_flags=
  	compiler.cpp.extra_flags=-mcall-prologues -fno-split-wide-types -finline-limit=3 -ffast-math
  	compiler.ar.extra_flags=
  	compiler.objcopy.eep.extra_flags=
  	compiler.elf2hex.extra_flags=


    Default PIN on the  Mega2560 Pin 37 (PORTC0)

    To change SWI pin, open file atca_hal.h in libraries/cryptoauthlibmega and define the following
   
    #define   SWI_PORT    PORTC
    #define   SWI_INPUT_PIN   PINC
    #define   SWI_DDR     DDRC
    #define   SWI_PIN     PORTC0
`

Here is the [file](https://github.com/PhillyNJ/ATMEGA/blob/master/CryptAuthLibAvrMega.zip)

**Comment by el-samiyel** [comment]
Wow, I will try this out the Mega. Apologies, I should have been more clear on my original post, I'm not used to getting such quick responses.

What I meant to say was 'ported to Arduino SAMD21 for the zero': (SAMD21/crypto/)

Essentially, I am after a library that is able to carryout AES encryption/decryption between two 32bit Arduino zero. 

I found a lib that's able to do this, but only on [8bit AVR](https://github.com/DavyLandman/AESLib). I came across your GitHub as the SAMD21 was mentioned.

Any info will be much appreciated!
